### ISSUE TITLE
Feature Request: withNamedDefault

### ISSUE THREAD

**Issue by tenowg** [inversify/InversifyJS#279]
## Expected Behavior

When multiple bindings are named/tagged will return a default binding if no name/tag is given. If no default binding is found there would be no change in current behavior (throw ambiguous match found)
## Current Behavior

Throws ambiguous error when attempt to inject.
## Possible Solution

Add a function similar to whenTagetNameIsDefault();
## Context

For the rare time that a default class will work for a set of named classes if a name is not supplied.

My personal use case, I am generating a Handler class for use on server side, and a client class (still used on server side, but only when a client requests it), it should be bound to the same identifier, the client would be named so when a client initiates the call it gets a fresh copy of the class, but the handler class would be injected into classes on the server, be singleton, but be able to be called using the same identifier, but without needing to know the generated name created by the server.

```
kernel.bind<Class>(Class).to(Class).whenTargetNamed("client");
kernel.bind<Class>(Class).to(HandlerClass).whenTargetNamedIsDefault();

@inject(Class) class: Class // injects default
@inject(Class) @named("client") // injects client class.
```

**Comment by remojansen** [comment]
I think is a good ideal. Will work on this after the 2.0 release :)

**Comment by remojansen** [comment]
@tenowg 2.0.0 has been released expect this coming soon!

**Comment by remojansen** [comment]
I though this was going to be a small thing but it is turning out to be quite complex :sob: I'm nearly done but it introduces some breaking changes so it will be released as 3.0.0. It took forever to release 2.0.0 and only one week to release 3.0.0 :smile:

**Comment by remojansen** [comment]
Good news :tada: https://github.com/inversify/InversifyJS/pull/392

**Comment by remojansen** [comment]
Default target will be available in 3.0.0 docs available at https://github.com/inversify/InversifyJS/blob/master/wiki/default_targets.md
